aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+    });
+    //
+}
+```
+
+### Options
+
+You can customize various options such as the header and cookie names used by antiforgery when you include it in `ConfigureServices`:
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddAntiforgery(options => 
+{
+  options.CookieDomain = "mydomain.com";
+  options.CookieName = "X-CSRF-TOKEN-COOKIENAME";
+  options.CookiePath = "Path";
+  options.FormFieldName = "AntiforgeryFieldname";
+  options.HeaderName = "X-CSRF-TOKEN-HEADERNAME";
+  options.RequireSsl = false;
+  otpions.SuppressXFrameOptionsHeader = false;
+});
+```
+
+<!-- QAfix fix table -->
+
+|Option        | Description |
+|------------- | ----------- |
+|CookieDomain  | The domain of the cookie. Defaults to null. |
+|CookieName    | The name of the cookie. If not set, the system will generate a unique name beginning with the `DefaultCookiePrefix` (".AspNetCore.Antiforgery."). |
+|CookiePath    | The path set on the cookie. |
+|FormFieldName | The name of the hidden form field used by the antiforgery system to render antiforgery tokens in views. |
+|HeaderName    | The name of the header used by the antiforgery system. If null, the system will consider only form data. |
+|RequireSsl    | Specifies whether SSL is required by the antiforgery system. Defaults to `false`. If `true`, non-SSL requests will fail. |
+|SuppressXFrameOptionsHeader  | Specifies whether to suppress generation of the `X-Frame-Options` header. By default the header is generated with a value of "SAMEORIGIN". Defaults to `false`. |
+
+### IAntiforgeryAdditionalDataProvider
+
+The `IAntiForgeryAdditionalDataProvider` type allows developers to extend the behavior of the anti-XSRF system by round-tripping additional data in each token. The `GetAdditionalData` method is called each time a field token is generated, and the return value is embedded within the generated token. An implementer could return a timestamp, a nonce, or any other value and then call `ValidateAdditionalData` to validate this data when the token is validated. The client's username is already embedded in the generated tokens, so there is no need to include this information. If a token includes supplemental data but no `IAntiForgeryAdditionalDataProvider` has been configured, the supplemental data is not validated.
+
+## Fundamentals
+
+CSRF attacks rely on built-in default browser behavior of sending cookies associated with a particular domain with every request to made to that domain. These cookies are stored within the browser and frequently include session cookies for authenticated users. Cookie-based authentication is by far the most popular form of authentication used by web applications. However, token-based authentication systems have been growing in popularity in recent years, especially for SPAs and other "smart client" scenarios.
```

Comment:
Cookie-based authentication is ~~by far~~ the most popular form of authentication used by web applications.